### PR TITLE
adding tests for groups in v2 APIs

### PR DESCRIPTION
**Pull request by ctschacher**
Adding tests for groups in v2 APIs. 

- Add 1 group 
- add multiple groups 
- delete group

https://gravitee.atlassian.net/browse/GKO-716

### CODE REVIEW

File: test/integration/apidefinition/v2/create_withContext_andMultipleGroups_test.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Create", labels.WithContext, func() {
+	timeout := constants.EventualTimeout
+	interval := constants.Interval
+
+	ctx := context.Background()
+
+	It("should create one group", func() {
```
Comment (a-cordier) [inline test/integration/apidefinition/v2/create_withContext_andMultipleGroups_test.go:39]: I guess this should be updated to reflect the fact that there are more than one group ?

Also, we don't create a group here, do we ? The operator simply attaches the groups to the API, maybe we should think about another title, especially given that group creation with the operator might become an actual thing.

Comment (a-cordier) [reply test/integration/apidefinition/v2/create_withContext_andMultipleGroups_test.go:39]: As an addition, that's a good idea to attach several groups at once, but how about melting this in another test about group deletion ?

the test would create 2 or 3 groups and remove one of them, then assert that APIM has been updated as it should.

WDYT ?

File: test/integration/apidefinition/v2/create_withContext_andMultipleGroups_test.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+		Eventually(func() error {
+			apiExport, err := apim.Export.V2Api(fixtures.API.Status.ID)
+			if err != nil {
+				return err
+			}
+
+			exportedGroups := apiExport.Spec.Groups
+			sort.Strings(exportedGroups)
+			return assert.Equals("groups", expectedGroups, exportedGroups)
```
Comment (a-cordier) [inline test/integration/apidefinition/v2/create_withContext_andMultipleGroups_test.go:71]: You can use
```golang
assert.SliceEqualsSorted("groups", expectedGroups, exportedGroups)
```

File: test/integration/apidefinition/v2/create_withContext_andOneGroup_test.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+		Eventually(func() error {
+			apiExport, err := apim.Export.V2Api(fixtures.API.Status.ID)
+			if err != nil {
+				return err
+			}
+
+			exportedGroups := apiExport.Spec.Groups
+
+			return assert.Equals("groups", expectedGroups, exportedGroups)
```
Comment (a-cordier) [inline test/integration/apidefinition/v2/create_withContext_andOneGroup_test.go:65]: Why don't we sort there ?

File: test/integration/apidefinition/v2/create_withContext_andMultipleGroups_test.go (hunk 2 above)
Comment (ctschacher) [reply test/integration/apidefinition/v2/create_withContext_andMultipleGroups_test.go:71]: Oh, nice one, thanks!

**Review by a-cordier — APPROVED** [review]
👍

**Pull request merged**